src/main/java/com/siit/hospital_manager/repository/AppointmentRepository.java

```diff
@@ -0,0 +1,37 @@
+package com.siit.hospital_manager.repository;
+
+import com.siit.hospital_manager.model.Appointment;
+import com.siit.hospital_manager.model.AppointmentStatus;
+import com.siit.hospital_manager.model.Doctor;
+import com.siit.hospital_manager.model.Patient;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.*;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface AppointmentRepository extends JpaRepository<Appointment, Integer> {
+    List<Appointment> findAllByPatientId(Integer id);
+    List<Appointment> findAllByPatientIdAndAppointmentStatus(Integer id, AppointmentStatus appointmentStatus);
+    List<Appointment> findAllByDoctorIdAndAppointmentStatus(Integer id, AppointmentStatus appointmentStatus);
```

Comment:
line 20 and 21 do the same thing, check if you can group these queries and have less similar code